### ISSUE TITLE
CKV_AWS_103: Check "Ensure that Application Load Balancer Listener is using TLS v1.2" produces false positives with network load balancers

### ISSUE THREAD

**Issue by ankon** [bridgecrewio/checkov#2928]
**Describe the issue**

Check sources https://github.com/bridgecrewio/checkov/blob/master/checkov/cloudformation/checks/resource/aws/ALBListenerTLS12.py

**Examples**

```yaml
  AppLB:
    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
    Properties:
      Scheme: internet-facing
      Type: network                      # <<<<<<<<<<<<<<<<
      Subnets: !Ref Subnets
      LoadBalancerAttributes:
        - Key: access_logs.s3.enabled
          Value: true
        - Key: access_logs.s3.bucket
          Value: THE_BUCKET
      Tags:
        - Key: Name
          Value: !Ref AWS::StackName

  AppHTTPSListener:
    Type: AWS::ElasticLoadBalancingV2::Listener
    Properties:
      LoadBalancerArn: !Ref AppLB
      Port: 443
      Protocol: TLS
      Certificates:
        - CertificateArn: !Ref AcmCertificateArn
      AlpnPolicy:
        - HTTP1Only
      DefaultActions:
        - Type: forward
          TargetGroupArn: !Ref AppTargetGroup
      # NOTE: https://docs.aws.amazon.com/elasticloadbalancing/latest/network/create-tls-listener.html
      SslPolicy: ELBSecurityPolicy-TLS13-1-2-2021-06
```

This produces:
~~~~
Check: CKV_AWS_103: "Ensure that Application Load Balancer Listener is using TLS v1.2"
	FAILED for resource: AWS::ElasticLoadBalancingV2::Listener.AppHTTPSListener
	File: THE_FILE
	Guide: https://docs.bridgecrew.io/docs/bc_aws_general_43

		222 |   AppHTTPSListener:
		223 |     Type: AWS::ElasticLoadBalancingV2::Listener
		224 |     Properties:
		225 |       LoadBalancerArn: !Ref AppLB
		226 |       Port: 443
		227 |       Protocol: TLS
		228 |       Certificates:
		229 |         - CertificateArn: !Ref AcmCertificateArn
		230 |       AlpnPolicy:
		231 |         - HTTP1Only
		232 |       DefaultActions:
		233 |         - Type: forward
		234 |           TargetGroupArn: !Ref AppTargetGroup
		235 |       # NOTE: https://docs.aws.amazon.com/elasticloadbalancing/latest/network/create-tls-listener.html
		236 |       SslPolicy: ELBSecurityPolicy-TLS13-1-2-2021-06
~~~~

**Version (please complete the following information):**
 - Checkov Version 2.0.1095

**Additional context**

NLBs support TLS1.3 policies (https://aws.amazon.com/about-aws/whats-new/2021/10/aws-network-load-balancer-supports-tls-1-3/), but the check only looks at the listener and not at the configured type of the load balancer.

This then also produces a misleading output message (it talks about "Application Load Balancer"); the documentation at https://docs.bridgecrew.io/docs/bc_aws_general_43 actually refers to it as "AWS Load Balancer".

**Comment by gruebel** [comment]
hi @ankon you totally right, this check doesn't include the new TLS 1.3 policies yet. Are you interested in contributing the needed change and add 1-2 test cases? You already found the correct Python file to do the change and then would just need to add the test case 🙂 

we should accept following new policies
- starts with `ELBSecurityPolicy-TLS13-1-2`
- starts with `ELBSecurityPolicy-TLS13-1-3`

**Comment by ankon** [comment]
I actually had a look :)

My biggest question: How would I check the type of the associated LoadBalancer resource? Right now the TLS13 policies aren't available for ALBs, only for NLBs.

**Comment by gruebel** [comment]
Nice, if you need more guidance, just reach out here :) 

no need, because the current code, just checks, if the protocol is `HTTPS` (ALB) or `TLS` (NLB) and then looks for the policy. So, you just need to extend the `startswith((..., "ELBSecurityPolicy-TLS13-1-2", "ELBSecurityPolicy-TLS13-1-3"))`
https://github.com/bridgecrewio/checkov/blob/85c0bef29929a45ad57530010aceed96679b765b/checkov/cloudformation/checks/resource/aws/ALBListenerTLS12.py#L28-L32

**Comment by ankon** [comment]
@gruebel turned out to indeed not be that hard ... https://github.com/bridgecrewio/checkov/pull/2934 :)
